gatorgrade/generate/generate.py

```diff
@@ -1,21 +1,82 @@
-"""Generate a GatorGrade configuration file.
+"""Generate a YAML file with default messages and specific paths."""
+import os
+from typing import List
 
-Gatorgrade file will include paths to whitelisted files
-and default GatorGrader checks.
-"""
+# Define colors for terminal output
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
 
-# Import the necessary libraries
 
-
-def create_targeted_paths_list():
+def create_targeted_paths_list(
+    key_word_list: List[str], relative_run_path: str = "."
+) -> List[str]:
     """Generate a list of targeted paths by walking the paths."""
+    targeted_paths = []
     # Go through the root repo, the sub dictionaries and files.
-    # Select only files in the dictionaries with specific names.
-    # Add those targeted file paths into a list and returns it.
+    # The os.walk will only scan the paths.
+    # So the empty folders containing nothing won't be gone through
+    for dirpath, _, filenames in os.walk(relative_run_path):
+        # Split path string into multiple layers of directories
+        path_dir_list = dirpath.split("/")
+        # Ignore folder starting with double underscore
+        if any(path.startswith("__") for path in path_dir_list):
+            continue
+        # Ignore hidden folders and first layer. the root repo is always dot
+        # Keep double dot. It represents the returning sign
```

Comment:
Hey! 👋 Can you give me more context on the comment "It represents the returning sign". This comment to me means that somehow if you come across a `..`, your function terminates?

Reply:
Hi, thanks for your review. Here `..`is used the same with the terminal. Since `os.walk` will treat the current repo as the current path. So if you want to check the directories before the current one, you'd better to go to the last repo by using `..`

Reply:
@Yanqiao4396 thanks for the explanation!